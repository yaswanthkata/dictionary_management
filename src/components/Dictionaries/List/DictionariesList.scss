@import "../../../styles/variables";
.dictionaries-list {
    list-style-type: none;
    height: 400px;
    padding-top:10px; 
    overflow: auto;
    li {
        padding: 10px;
        overflow: auto;
        img {
          float: left;
          margin: 0 15px 0 0;
        }
      }
      
      // li:hover {
      //   background: $color-border-light;
      //   cursor: pointer;
      // }
      li.selected {
        background:$color-border-dark;
      }
      .title{
          text-align: start;
          padding: 10px;
          button{
            float: right;
          }
      }
  }
 
  @media screen and (max-width: 600px){
    .dictionaries-list {
      height: 200px;
    }
  }