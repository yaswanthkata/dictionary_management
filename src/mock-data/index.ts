import { Dictionary, DictionaryRow } from "./../types";

export const rows: DictionaryRow[] = [
    { id: "1", domain: "Stonegrey", range: "Dark Grey", errors: [] },
    { id: "2", domain: "Stonegrey", range: "Dark Grey", errors: [] }
];
export const dictionaries: Dictionary[] = [
    { id: "1", name: "Yaswanth kata", rows: [] },
    { id: "2", name: "OneDot", rows: [] }
];
