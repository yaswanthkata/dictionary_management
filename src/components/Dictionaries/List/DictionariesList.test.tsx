import React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DictionariesList, { IDictionariesListProps } from './index';
import { dictionaries } from '../../../mock-data';

configure({ adapter: new Adapter() });


const props: IDictionariesListProps = {
    dictionaries: dictionaries,
    selectedDictionary: null,
    onSelect: jest.fn(),
    onDelete: jest.fn()
}

let componentWrapper: ShallowWrapper<React.Component>;

beforeEach(() => {
    componentWrapper = shallow(<DictionariesList {...props} />);
});


it('should render dictionaries list component correctly', () => {
    expect(componentWrapper).toMatchSnapshot();
});

it('should render one <ul> element', () => {
    expect(componentWrapper.find('ul')).toHaveLength(1);
});

it("should render  a list of dictionaries", () => {
    const listItems = componentWrapper.find('li').length;

    expect(listItems).toEqual(2);
});
