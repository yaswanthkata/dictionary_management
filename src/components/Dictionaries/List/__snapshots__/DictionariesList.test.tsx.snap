// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dictionaries list component correctly 1`] = `ShallowWrapper {}`;
